### PR TITLE
MM-69257: Address issues with github digest on Mattermost

### PR DESCRIPTION
**Pull request by jgheithcock**
## Summary

Fixes cases where the daily SLA review digest fails to post for large orgs (e.g. `mattermost` on hub) and makes the digest easier to configure and read.

- **Digest service user** — Adds a `Digest service user (Mattermost username)` plugin setting so admins can choose which connected GitHub account runs org-wide digest GraphQL and team lookups, instead of relying on the lexicographically-first connected user.
- **Message size cap** — Clips channel digest posts at 64K runes and appends a notice when truncated, so large org digests still post instead of failing silently at Mattermost post limits.
- **Bucket labels** — Replaces day-range headers (`1-3 days overdue`, etc.) with clearer labels such as `Overdue`, `Overdue by 1 week`, `Overdue by 1 month`, and `Over a year overdue`.

Also updates the overdue-reviews channel help text to reference the new service user setting.

## Ticket Link

https://mattermost.atlassian.net/browse/MM-69257

## Context

After the org-wide GraphQL digest landed (MM-68539), digests worked for small orgs but could fail for large ones: the service caller was unpredictable, and messages with hundreds of overdue entries exceeded post size limits (~65K runes observed for `mattermost`).

## Test plan

- [ ] Set **GitHub Organizations**, **PR review target (days)**, **Post overdue review alerts to channel**, and **Digest service user** to a Mattermost user connected via `/github connect`
- [ ] Confirm the bot is a member of the target channel
- [ ] Trigger or wait for the daily digest; verify it posts using the configured service user (check server logs for `SLA digest using configured service user`)
- [ ] For a large org, confirm the digest posts and shows `_This digest was clipped due to message size limits._` when content would exceed 64K runes
- [ ] Verify bucket headers use the new labels (`Overdue`, `Overdue by 1 week`, …)
- [ ] Leave **Digest service user** empty and confirm digest still runs using the previous lexicographic fallback
- [ ] Set **Digest service user** to a user who is not connected to GitHub; confirm digest does not post and logs a warning (day marker should not advance until a successful scan)

## Change Impact: 🟡 Medium

**Reasoning:** The changes introduce configurable service user selection and message size constraints for the digest feature with proper test coverage. While the modifications affect a user-facing digest operation and change behavioral contracts (service user selection logic and bucket labeling), they are well-contained to the digest subsystem with clear fallback mechanisms and comprehensive test additions.

**Regression Risk:** Moderate. The service user selection logic shifts from lexicographic ordering to explicit configuration with tested fallback behavior, which changes existing operation but includes safeguards. Message clipping introduces a new truncation path for large digests that could alter output appearance, though visual truncation markers provide transparency. Bucket label changes are cosmetic but affect digest parsing. The refactoring of `truncatePostMessage` logic to a new `truncateMessageAtRunes` utility is low-risk and includes dedicated test coverage.

**QA Recommendation:** Medium to High manual QA recommended. Prioritize testing: (1) digest service user selection logic when configured, unconfigured, and when selected user lacks GitHub connectivity; (2) large digest message clipping at rune boundaries (especially multi-byte characters) and truncation notice appearance; (3) new bucket label formatting in actual digest output; and (4) fallback to lexicographic ordering when service user is unspecified. Focus on edge cases like UTF-8 boundary conditions during clipping. Manual QA can be reduced only if comprehensive end-to-end digest integration tests exist and cover all three feature areas.

### CODE REVIEW

File: plugin.json [hunk 1]
```diff
@@ -146,7 +146,13 @@
                 "key": "OverdueReviewsChannelID",
                 "display_name": "Post overdue review alerts to channel (ID):",
                 "type": "text",
-                "help_text": "Optional. Paste a channel ID (Channel menu > View Info). When set together with PR review target (days) and GitHub Organizations, the plugin posts one aggregated overdue-review digest for those orgs to this channel each day shortly after midnight in the server local timezone. Requires at least one user connected via /github connect. The GitHub bot must be a member of the channel."
+                "help_text": "Optional. Paste a channel ID (Channel menu > View Info). When set together with PR review target (days) and GitHub Organizations, the plugin posts one aggregated overdue-review digest for those orgs to this channel each day shortly after midnight in the server local timezone. Requires digest service user (below) to be connected via /github connect. The GitHub bot must be a member of the channel."
```
[the remark this answers is not in the document]
Comment (jgheithcock) [reply plugin.json:149]: The help text literally says "Optional." at the front, following the other help text patterns.

**Review by nang2049 — APPROVED** [review]
LGTM, please merge with master as we have merged MM-69266 org-webhook fix :)
